### PR TITLE
Update spectraray importer

### PR DESCRIPTION
**Pull request by MarJMue**
Changes the spectraray importer, to use the same data structure as the nexus one. 

Changes the examples and tests accordingly.

### CODE REVIEW

File: src/elli/importer/spectraray.py [hunk 1]
```diff
@@ -8,48 +8,69 @@
 
 
 def read_spectraray_psi_delta(
-    fname: str, sep: str = r"\s+", decimal: str = "."
+    fname: str, sep: str = " ", decimal: str = "."
```
Comment (domna) [inline src/elli/importer/spectraray.py]: Why not use any whitespace as separator? My gut feeling is that it's a little bit more resilient against formatting errors in files. Or did you encounter errors with `\s+`?

File: src/elli/importer/spectraray.py [hunk 2]
```diff
@@ -8,48 +8,69 @@
...
+        psi_delta_df.loc[:, "Δ"] <= 180, psi_delta_df.loc[:, "Δ"] - 360
     )
 
-    return psi_delta
+    return psi_delta_df
 
 
 def read_spectraray_mmatrix(
-    fname: str, sep: str = r"\s+", decimal: str = "."
+    fname: str, sep: str = r" ", decimal: str = "."
```
Comment (domna) [inline src/elli/importer/spectraray.py]: Same as above.

File: src/elli/importer/spectraray.py [hunk 3]
```diff
@@ -83,15 +104,15 @@ def read_spectraray_mmatrix(
 
 
 def read_spectraray_rho(
-    fname: str, sep: str = r"\s+", decimal: str = "."
+    fname: str, sep: str = r" ", decimal: str = "."
```
Comment (domna) [inline src/elli/importer/spectraray.py]: Same as above.

**Review by domna — APPROVED** [review]
I assume the formatting changes in the notebooks come from a newer jupyter version, right?

**Comment by MarJMue** [comment]
> I assume the formatting changes in the notebooks come from a newer jupyter version, right?

I think so.

File: src/elli/importer/spectraray.py (hunk 1 above)
Comment (MarJMue) [reply src/elli/importer/spectraray.py]: I wanted to parse the header with str.split() and afaik that does not work with r-strings? But I could not find any good information on that.

Comment (domna) [reply src/elli/importer/spectraray.py]: Ah, the problem is not the raw string, but rather that str.split does not evaluate regexes. You could use `re.split`. I'll put a suggestion below

File: src/elli/importer/spectraray.py [hunk 4]
```diff
@@ -8,48 +8,69 @@
...
     )
+    psi_delta_df.dropna(axis="columns", how="all", inplace=True)
+
+    # index data correctly
+    psi_delta_df.index.name = "Wavelength"
+
+    with open(fname) as f:
+        header = f.readlines()[0]
+
+    aois = list(map(float, header.split(sep)[2::2]))
```
Comment (domna) [inline src/elli/importer/spectraray.py]:
```suggestion
    aois = list(map(float, re.split(sep, header)[2::2]))
```

**Comment by domna** [comment]
> > I assume the formatting changes in the notebooks come from a newer jupyter version, right?
> 
> I think so.

Seems nice. Looks like they are working on having better git diffs for the notebooks :)

File: src/elli/importer/spectraray.py (hunk 4 above)
Comment (MarJMue) [reply src/elli/importer/spectraray.py]: I tried that before, but now i have found my mistake :+1:

**Pull request merged**